cluster/cluster.yaml

```diff
@@ -2006,6 +2006,101 @@ Resources:
             Resource: "*"
             Action:
             - "kms:Decrypt"
+  IdentityProvider:
+    Type: Custom::IdentityProvider
+    Properties:
+      ServiceToken: !GetAtt IdentityProviderCreator.Arn
+      Region: !Ref "AWS::Region"
+      ProviderName: "{{ .Cluster.LocalID }}.{{ .Values.hosted_zone }}"
+  IdentityProviderCreator:
+    Type: AWS::Lambda::Function
+    Properties:
+      Runtime: python2.7
+      Handler: index.lambda_handler
+      MemorySize: 128
+      Role: !GetAtt LambdaExecutionRole.Arn
+      Timeout: 30
+      Code:
+        ZipFile: !Sub |
+          import boto3
+          from botocore.exceptions import ClientError
+          import json
+          import cfnresponse
+
+          iam = boto3.client("iam")
+
+          def create_provider(provider_name):
+            try:
+              resp = iam.create_open_id_connect_provider(
+                  Url= "https://" + provider_name,
+                  ClientIDList=['sts.amazonaws.com'],
+                  ThumbprintList=['9e99a48a9960b14926bb7f3b02e22da2b0ab7280']
+              )
+              return(True, resp['OpenIDConnectProviderArn'])
+            except Exception as e:
+              return (False, "Cannot create OpenID Connect provider: " + str(e))
+
+          def delete_provider(arn):
+            try:
+              resp = iam.delete_open_id_connect_provider(OpenIDConnectProviderArn=arn)
+              return (True, "OpenID Connect provider with ARN " + arn + " deleted")
+            except ClientError as e:
+              if e.response['Error']['Code'] == "NoSuchEntity":
+                # no need to delete a thing that doesn't exist
+                return (True, "OpenID Connect provider with ARN " + arn + " does not exist, deletion succeeded")
+              else:
+                return (False, "Cannot delete OpenID Connect provider with ARN " + arn + ": " + str(e))
```

Comment:
That's not very Pythonic, just throw an exception and catch it later.

Reply:
How do you mean? Like with nested `try`s?

Reply:
I've meant using multiple return of a bool + string instead of just letting the exception propagate upwards, this looks like something from Go. I don't see why the `try` blocks would even need to be nested, you can just have a `try/catch` around the `delete` part and parse the error reason, or just not return any string for the happy path.